### ISSUE TITLE
Weird visual corruption issue with dash in Top position

### ISSUE THREAD

**Issue by dimitri-vanlanduyt** [G-dH/vertical-workspaces#202]
I prefer placing the dock in the top position and I have been using this wonderful extension for that. 
However, since a few weeks, the rendering of the dock in that specific position has started suffering from a form of "visual corruption" (see screenshots), both in the rendering of the app icons, and in the tooltips below. 
It appears to first render them correctly and then it changes ( I can sometimes see the change happening).
This is consistent. 
I experienced the issue in Gnome 3.36 (F40) and now also in Gnome 3.37 (F41). 

I tried disabling all other plugins and the issue persists. It does not occur with the dock in a different position.

Weirdly, this corruption seems to happen along a line that gradually decreases (see final screenshot and compare icon per icon and where the shift is happening in the Y plane: this decreases). 
Perhaps this observation this may give a hint towards the root cause, most probably not.

**To Reproduce**
Visualise the dock

**Expected behavior**
No visual corruption

**Screenshots**
![Screenshot From 2024-10-31 07-32-16](https://github.com/user-attachments/assets/2c387837-399f-41f2-85a0-0305b652ba06)
![Screenshot From 2024-10-31 07-26-19](https://github.com/user-attachments/assets/8978b796-9de2-43de-b0f7-ba446372a0e3)
![Screenshot From 2024-10-31 07-40-50](https://github.com/user-attachments/assets/19a51929-5122-43f5-89fe-4b238a35745d)

**System:**
 - OS: Fedora 40, 41
 - Version of V-Shell extension: 47/70
 - Source of installation: extension.gnome.org

**Comment by dimitri-vanlanduyt** [comment]
It also occurs in the bottom position, but not when the dock is rendered vertically.

**Comment by G-dH** [comment]
Have you tried reproducing the issue with V-Shell disabled?

**Comment by dimitri-vanlanduyt** [comment]
Yes, without the V-Shell extension, the dock renders perfectly fine.

**Comment by dimitri-vanlanduyt** [comment]
I just noticed that it only occurs when the dock includes icons of running apps that are not pinned. 

On a workspace that only has pinned apps open , the corruption does not occur.

![Screenshot From 2024-10-31 08-11-49](https://github.com/user-attachments/assets/9f993f97-3da4-4008-85aa-90e6ad60e870)

**Comment by G-dH** [comment]
Wayland or X11?
Graphics driver?
Are you using any custom theme?
Does disabling the dash module in the V-Shell Settings help?

**Comment by dimitri-vanlanduyt** [comment]
Wayland/amdgpu. 
No custom theme. 
Turning off the dash module put the dock to the bottom and does not have the corruption issue (but also not the isolation in workspaces feature). 

I do isolate workspaces (so the dock changes when I switch workspace), and I notice that it doesn't occur when I turn that off. We are getting closer

**Comment by G-dH** [comment]
> I do isolate workspaces (so the dock changes when I switch workspace), and I notice that it doesn't occur when I turn that off.

Here's a refined version:

I noticed in your previous post that you're using this option and tested it myself, but it doesn't cause the issue on my system. In any case, V-Shell doesn't control dash rendering, so we can only hope to find a workaround if I can reproduce the issue.

**Comment by G-dH** [comment]
You can try this, when overview is open and the glitch is visible:
- open looking glass (Alt+F2 > lg > Enter)
- type `Main.overview.dash.hide()` and hit Enter
- type `Main.overview.dash.show()` and hit Enter
- press Escape key to close the lg
Can you still see the glitch?

**Comment by dimitri-vanlanduyt** [comment]
Yes, the glitch is still visible after that.

**Comment by G-dH** [comment]
How about switching workspaces in the overview? Are the icons still being recreated with the glitch?

**Comment by nsurvance** [comment]
I wanted to hop in here because I am also having this same issue, but I keep my dock positioned horizontally at the bottom.
![image](https://github.com/user-attachments/assets/65b8f311-9c26-46df-9432-653bb68e3760)

I similarly am having this issue on Gnome 45 (F40) and 46 (F41).
V-Shell Version 70
Installed from Extensions Manager
AMD, Wayland
and no other themes installed

I do **NOT** isolate work spaces though.  Mine is consistent acrossed workspaces and the visual bug appears across all of them as long as there is an unpinned app loaded--

https://github.com/user-attachments/assets/c966ae06-4b7a-46dc-bac0-57d4e04f2761

Issue still occurs with no other extensions on--
![image](https://github.com/user-attachments/assets/037f1d05-7e77-4d35-91ab-17783e2c72a2)

And the problem only occurs when Vshell is enabled--

https://github.com/user-attachments/assets/a84215ad-3855-4954-835b-d077084e2c41

I can also confirm that the issue disappears when no non-pinned apps are open.
And disabling/enabling the dash in looking glass didn't work for me either :(

If it helps, I believe my settings are unchanged from my previous bug report except now I have the background blur enabled:
https://github.com/G-dH/vertical-workspaces/issues/182#issuecomment-2220901223

**Comment by G-dH** [comment]
It seems the horizontal position of the dock might not be rounded.

**Comment by G-dH** [comment]
When the overview is open and the glitch is visible, please open the looking glass (Alt+F2 > lg > Enter), execute 
`Main.overview.dash.allocation.x1`
`Main.overview.dash.allocation.x2`
`Main.overview.dash.allocation.y1`
`Main.overview.dash.allocation.y2`
and show me the outputs.

**Comment by dimitri-vanlanduyt** [comment]
![Screenshot From 2024-11-04 08-35-29](https://github.com/user-attachments/assets/5a5d1768-8628-44aa-9be9-dc5014fe0d79)

x1 = 394.5
y1 = 44
x2 = 1795.5
y2 = 156

**Comment by G-dH** [comment]
Thanks. Can you install the V-Shell from the repository and let me know if the fix helped?

**Comment by dimitri-vanlanduyt** [comment]
It solved the issue!!

**Comment by G-dH** [comment]
Thank you very much @dimitri-vanlanduyt! I really appreciate your feedback and support.

**Comment by dimitri-vanlanduyt** [comment]
Perhaps it is unrelated, but I notice a similar corruption on the window captions in the secondary display. This is consistent. 

![Image](https://github.com/user-attachments/assets/62375152-04d2-4b4d-a4b6-985803789806)
